File: 2019/cpp/day_01a.cpp
#include <fstream>
#include <iostream>
#include <iterator>
#include <string>
#include <vector>

int main(int argc, char* argv[]) {
  // Get input
  std::string input = "../input/day_01_input";
  if (argc > 1) {
    input = argv[1];
  }
  std::ifstream file(input);
  std::istream_iterator<int> start(file), end;
  const std::vector<int> masses(start, end);

  // Solve
  long long total_fuel = 0;
  constexpr int divisor = 3;
  constexpr int subtract = 2;
  for (const auto& mass : masses) {
    total_fuel += mass / divisor - subtract;
  }

  // STL variant
  // long long total_fuel = std::accumulate(start, end, 0, [](long long
  // sum_fuel, const int ele) { return sum_fuel + getRecursiveFuel(ele);} );

  std::cout << total_fuel << '\n';
  return total_fuel;
}

File: 2019/python/day_01a.py
#!/usr/bin/python3

import sys


def main():
    # Get input
    file_name = "../input/day_01_input"
    if len(sys.argv) > 1:
        file_name = sys.argv[1]
    file = open(file_name)
    input = [int(ele) for ele in file]

    # Solve
    total_fuel = 0
    for mass in input:
        total_fuel = total_fuel + int(mass / 3) - 2
    print(total_fuel)
    return total_fuel


if __name__ == "__main__":
    main()
